cmd/tpmk/sshclient.go

```diff
@@ -194,12 +197,29 @@ func runSSHClient(opt sshClientOptions, args []string) error {
 		HostKeyCallback: hostKeyCallback,
 	}
 
-	// Connect to the SSH server and start a session
-	client, err := ssh.Dial("tcp", host, config)
-	if err != nil {
-		return errors.Wrap(err, "connecting to "+host)
+	var client Client
+	if opt.proxySocks5 != "" {
+		socksConnection, err := proxy.SOCKS5("tcp", opt.proxySocks5, nil, proxy.Direct)
+		if err != nil {
+			return errors.Wrap(err, "could not create socks5 proxy "+opt.proxySocks5)
+		}
+		conn, err := socksConnection.Dial("tcp", host)
+		if err != nil {
+			return errors.Wrap(err, "could not create connection with socks.")
+		}
+		clientConnection, chans, reqs, err := ssh.NewClientConn(conn, host, config)
+		if err != nil {
+			return errors.Wrap(err, "could not create socks5 proxy client "+opt.proxySocks5)
+		}
+		client = ssh.NewClient(clientConnection, chans, reqs)
+		defer client.Close()
+	} else {
+		client, err = ssh.Dial("tcp", host, config)
+		if err != nil {
+			return errors.Wrap(err, "connecting to "+host)
+		}
+		defer client.Close()
 	}
-	defer client.Close()
```

Comment:
You should be able to keep this `defer` after the `}` -- if you couldn't, that implies a variable scoping problem.